raiden_contracts/tests/test_channel_cooperative_settle.py

```diff
@@ -0,0 +1,409 @@
+    deposit_A = 0
+    deposit_B = 0
+    balance_A = 0
+    balance_B = 0
+
+    channel_identifier = create_channel_and_deposit(A, B, deposit_A, deposit_B)
+
+    # Create cooperative settle signatures
+    (signature_A, signature_B, signature_C) = create_cooperative_settle_signatures(
+        [A, B, C],
+        channel_identifier,
+        A, balance_A,
+        B, balance_B
+    )
+
+    pre_account_balance_A = custom_token.call().balanceOf(A)
+    pre_account_balance_B = custom_token.call().balanceOf(B)
+    pre_balance_contract = custom_token.call().balanceOf(token_network.address)
+
+    token_network.transact({'from': C}).cooperativeSettle(
+        channel_identifier,
+        A, balance_A,
+        B, balance_B,
+        signature_A,
+        signature_B
+    )
+
+    cooperative_settle_state_tests(
+        channel_identifier,
+        A, balance_A,
+        B, balance_B,
+        pre_account_balance_A,
+        pre_account_balance_B,
+        pre_balance_contract
+    )
+
+
+def test_cooperative_settle_channel_state(
+        web3,
+        custom_token,
+        token_network,
+        create_channel_and_deposit,
+        get_accounts,
+        create_cooperative_settle_signatures,
+        cooperative_settle_state_tests
+):
+    (A, B, C) = get_accounts(3)
+    deposit_A = 20
+    deposit_B = 10
+    balance_A = 5
+    balance_B = 25
+
+    # Create channel and deposit
+    channel_identifier = create_channel_and_deposit(A, B, deposit_A, deposit_B)
+
+    # Create cooperative settle signatures
+    (signature_A, signature_B) = create_cooperative_settle_signatures(
+        [A, B],
+        channel_identifier,
+        A, balance_A,
+        B, balance_B
+    )
+
+    pre_account_balance_A = custom_token.call().balanceOf(A)
+    pre_account_balance_B = custom_token.call().balanceOf(B)
+    pre_balance_contract = custom_token.call().balanceOf(token_network.address)
+
+    # Settle the channel
+    token_network.transact({'from': C}).cooperativeSettle(
+        channel_identifier,
+        A, balance_A,
+        B, balance_B,
+        signature_A,
+        signature_B
+    )
+
+    cooperative_settle_state_tests(
+        channel_identifier,
+        A, balance_A,
+        B, balance_B,
+        pre_account_balance_A,
+        pre_account_balance_B,
+        pre_balance_contract
+    )
+
+
+def test_cooperative_settle_channel_wrong_balances(
+        web3,
+        custom_token,
+        token_network,
+        create_channel_and_deposit,
+        get_accounts,
+        create_cooperative_settle_signatures,
+        cooperative_settle_state_tests
+):
+    (A, B, C) = get_accounts(3)
+    deposit_A = 20
+    deposit_B = 10
+    balance_A = 7
+    balance_B = 23
+
+    balance_A_fail1 = 20
+    balance_B_fail1 = 11
+    balance_A_fail2 = 6
+    balance_B_fail2 = 8
+
+    # Create channel and deposit
+    channel_identifier = create_channel_and_deposit(A, B, deposit_A, deposit_B)
+
+    # Create cooperative settle signatures
```

Comment:
pedantic: No reason for the comment as the function is very aptly named :)

Reply:
Yes, removing them. The comments were there first. Then came the rest.